lnwire/allow_htlc_request.go

```diff
@@ -0,0 +1,107 @@
+package lnwire
+
+import (
+	"fmt"
+	"io"
+)
+
+// AllowHTLCRequest is sent to Bob from Alice. Alice wants to know if Bob is able to
+// make htlc chain with some other node with a given amount.
+// This message is sent to all nodes in possible HTLC chain to ensure that they can
+// accept it.
+// It is used to validate/filter possible paths in routing.
+type AllowHTLCRequestMessage struct {
+	// Id of node to which payment may be requested in future
```

Comment:
Is this the ID of the final destination? If so it should be removed as its inclusion defeats the purpose of onion routing.